XIPs/xip-reply-content-type.md

```diff
@@ -0,0 +1,60 @@
+---
+xip: 
+title: Reply Content Type
+description: Content type for reply messages
+author: Kunal Mondal
+status: Draft 
+type: Standards Track
+category: XRC
+created: 2023-03-24
+---
+
+## Abstract
+
+This XRC proposes a new content type for messages that enriches the messaging experience. It attempts to include only the bare minimum for clients to determine how to display such messages in order to provide flexibility for more rich types in the future.
+
+## Motivation
+
+When it comes to having a comfortable conversation, replying to messages is an absolute no brainer as it gives users the freedom to converse without any miscommunications. In order to provide users a real messaging platform, XMTP needs to provide this capability.
+
+## Specification
+
+Proposed content type:
+
+```js
+{
+  authorityId: "xmtp.org"
+  typeId: "reply"
+  versionMajor: 0
+  versionMinor: 1
+}
+```
+
+The reply message MUST include the following parameters:
+
+```ts
+{
+  // The ID of the message that is being replied to
+  inReplyToID: string
+}
+```
+
+The content of the encoded message is an EncodedContent object, serialized in the protobuf format (since we're guaranteed to always have the ability to deserialize that.)
+
+The content type of the reply's `EncodedContent` MUST be allowed. By default, the only allowed content type is `ContentTypeText`, but additional types can be optionally allowed as well.
+
+## Backward compatibility
+
+Clients encountering messages of this type must already be able to deal with messages of an unknown content type, so whatever considerations they're making there should work here too.
```

Comment:
It sounds like there are no backward compatibility concerns with this proposal, then let's just say that with the fewest words possible.

Reply:
We should strongly encourage users of the Reply content type to use `fallbackText` whenever possible, given this is rolling out into a world where most clients do not support replies.

The codecs can help a bit here. If the child content has a fallback, we could bubble that up to make it the fallback of the parent. That way even if your client doesn't support Replies, you can still read the message. If the child content is of type text, we could also make that text the fallback of the parent.